IHP/IDE/Data/Controller.hs

```diff
@@ -41,9 +41,15 @@ instance Controller DataController where
 
     action ShowQueryAction = do
         connection <- connectToAppDb
-        let query = (param @Text "query")
-        when (query == "") $ redirectTo ShowDatabaseAction
-        rows :: [[DynamicField]] <- if isQuery query then PG.query_ connection (fromString (cs query)) else PG.execute_ connection (fromString (cs query)) >> return []
+        let queryText = param @Text "query"
+        when (queryText == "") $ redirectTo ShowDatabaseAction
+        let query = fromString $ cs queryText
+
+        queryResult :: Either PG.SqlError [[DynamicField]] <- if isQuery queryText then
+                (PG.query_ connection query <&> Right) `catch` (pure . Left)
+            else
+                PG.execute_ connection query >> pure (Right []) `catch` (pure . Left)
```

Comment:
I think there's a bug here. When i enter a non-query like `insert into x;`, I still get the haskell error page. For select queries it works.

Likely the `catch` is only matching the `pure (Right [])` expression. If add `()` it could be fixed likely:

```haskell
(PG.execute_ connection query >> pure (Right [])) `catch` (pure . Left)
```

Reply:
Good catch! Fixed